src/test/java/uk/gov/mca/beacons/service/accounts/AccountHolderControllerIntegrationTest.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.mca.beacons.service.accounts;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.web.reactive.function.BodyInserters;
+import uk.gov.mca.beacons.service.model.AccountHolder;
+import uk.gov.mca.beacons.service.repository.AccountHolderRepository;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureWebTestClient
+class AccountHolderControllerIntegrationTest {
+
+  @Autowired
+  private WebTestClient webTestClient;
+
+  @MockBean
+  private AccountHolderRepository accountHolderRepository;
```

Comment:
I'm not sure it's the best idea to mock this repository out within this test. I assume we have done this because we want to check that the JSON response from the API is the `AccountHolder` object that you create and tell the repository to return on line 63/64 which in turn matches the JSON in the fixtures directory?

A few points to note on why I wouldn't mock it out in this case:

1. The test very much knows the internal implementation of how the `CreateAccountHolderService` works as it knows that it calls through to the `AccountHolderRepository` repository and a specific method on that interface
2. The value in this integration test is that we are attempting to persist in the database and in mocking out the persistence layer we might mask future issues that might arise from things like: table names changing, db constraints, column names changing

If you want to check that the JSON object returned contains the values you posted you could use `JsonPathAssertions` to check that the values in the JSON object are what we expect, based on the object we sent in the `createAccountHolderRequest.json` - minus the ID, but you can add a check to ensure that the id is not empty!

You can use [JsonPath](https://github.com/json-path/JsonPath) expressions to specify fields in an object you want to check and then provide the values that you expect.

The best example to follow that we have is in the [BeaconsControllerIntegrationTest](https://github.com/mcagov/beacons-service/blob/main/src/test/java/uk/gov/mca/beacons/service/beacons/BeaconsControllerIntegrationTest.java#L132)

More than happy to help out pair on this!

Reply:
I think you're spot on for both those reasons 👌🏻 

Asserting on JsonPath wasn't the issue (I think comparing to a test fixture and using JsonPath achieve the same aim?). 

When mocking nothing, the first test would pass but then the next run of the same test would fail because of the `UNIQUE` constraint on the `auth_id` column in the table.  (As I type this, alarm bells are going off in my head that I chose to mock something because the tested behaviour was entirely as expected 🤦🏻 Muppet ). 

Also thought about generating a random UUID for the `auth_id` so that the UNIQUE constraint would be satisfied with each test, but that's a bit of a workaround in that the db state is still different at the start of each test run.

It'd be more expensive, but cheaper in the long run if we just had a teardown step that reset database state in between integration tests.  Is there an easy way to do this in Spring/Hibernate/Junit?  What do you think about [something like this](https://miensol.pl/clear-database-in-spring-boot-tests/) (looks a bit involved)?

Reply:
@issyjenkins thoughts?

Reply:
(By cheaper in the long run I mean: will avoid incredibly annoying state bleeding test bugs if as a rule we just always reset state between tests)

Reply:
If the plan is to use the actual DB then I think I prefer generating a random ID and either inserting it into our test data or using JSONPathAssertions to just check the fields that aren't the ID. Having a before or after step that clears out the database in our test suite feels wrong

Reply:
We've gone with randomising the `authId` 👍🏻